geomesa-compute/src/main/scala/org/locationtech/geomesa/compute/spark/GeoMesaSpark.scala

```diff
@@ -187,6 +188,224 @@ object GeoMesaSpark extends LazyLogging {
+                case (a: java.lang.Double) => a / count
+                case _ => throw new Exception(s"couldn't match $name")
+              }
+              sf.setAttribute("avg_" + name.substring(6), avg)
+            }
+          }
+          sf.setAttribute(key, coverName)
+          sf.setDefaultGeometry(broadcastedCoverMap.value.getOrElse(coverName, null))
+          Some(sf)
+        } else {
+          None
+        }
+      }
+    }
+  }
+
+  def reduceAndAggregate(keyedData: RDD[(String, SimpleFeature)], countable: Broadcast[Array[(Int, String)]]): RDD[(String, SimpleFeature)] = {
+
+    // Reduce features by their covering area
+    val aggregate = keyedData.reduceByKey((featureA, featureB) => {
+      import org.locationtech.geomesa.utils.geotools.Conversions.RichSimpleFeature
+
+      val aggregateSft = GeoMesaSparkKryoRegistrator.getType("aggregate")
+
+      val typeA = featureA.getType.getTypeName
+      val typeB = featureB.getType.getTypeName
+      // Case: combining two aggregate features
+      if (typeA == "aggregate" && typeB == "aggregate") {
+        // Combine the "total" properties
+        (featureA.getProperties, featureB.getProperties).zipped.foreach((propA, propB) => {
+          val name = propA.getName.toString
+          if (propA.getName.toString.startsWith("total_") || propA.getName.toString == "count") {
+            val sum = (propA.getValue, propB.getValue) match {
+              case (a: Integer, b: Integer) => a + b
+              case (a: java.lang.Long, b: java.lang.Long) => a + b
+              case (a: java.lang.Double, b: java.lang.Double) => a + b
+              case _ => throw new Exception("Couldn't match countable type.")
+            }
+            featureA.setAttribute(propA.getName, sum)
+          }
+        })
+        featureA
+      // Case: combining two regular features
+      } else if (typeA != "aggregate" && typeB != "aggregate") {
+        // Grab each feature's properties
+        val featurePropertiesA = featureA.getProperties.toSeq
+        val featurePropertiesB = featureB.getProperties.toSeq
+        // Create a new aggregate feature to hold the result
+        val featureFields = Seq("empty", featureA.geometry) ++ Seq.fill(aggregateSft.getTypes.size - 2)("0")
+        val aggregateFeature = ScalaSimpleFeatureFactory.buildFeature(aggregateSft, featureFields, featureA.getID)
+
+        // Loop over the countable properties and sum them for both geonames simple features
+        countable.value.foreach { case (index, clazz) =>
+          val propA = featurePropertiesA(index)
+          val propB = featurePropertiesB(index)
+          val valA = if (propA == null) 0 else propA.getValue
+          val valB = if (propB == null) 0 else propB.getValue
+
+          val sum = (valA, valB) match {
+            case (a: Integer, b: Integer) => a + b
+            case (a: java.lang.Long, b: java.lang.Long) => a + b
+            case (a: java.lang.Double, b: java.lang.Double) => a + b
```

Comment:
By moving the sum calculation into block of the null check like you suggested below, I get around this issue. I also better handle the vice versa case (when propB is null)